src/wasm2asm.h

```diff
@@ -578,6 +583,49 @@ void Wasm2AsmBuilder::addExports(Ref ast, Module* wasm) {
   ast->push_back(ValueBuilder::makeStatement(ValueBuilder::makeReturn(exports)));
 }
 
+void Wasm2AsmBuilder::addGlobal(Ref ast, Global* global) {
+  assert(global->init);
+
+  if (global->init->is<Const>()) {
```

Comment:
instead of these 2 lines, can do
```
if (auto* const_ = global->init->dynCast<Const>()) {
```
dynCast returns the pointer cast to the type you want, or nullptr if it's the wrong type.